tfhe/src/shortint/client_key/mod.rs

```diff
@@ -40,8 +40,16 @@ impl ClientKey {
     /// // Generate the client key:
     /// let cks = ClientKey::new(PARAM_MESSAGE_2_CARRY_2);
     /// ```
-    pub fn new(parameters: Parameters) -> ClientKey {
-        ShortintEngine::with_thread_local_mut(|engine| engine.new_client_key(parameters).unwrap())
+    pub fn new<P>(parameters: P) -> ClientKey
```

Comment:
Shouldn't this still only accept `PBSParameters` ?

Are WopbsParam meant to be used / compatible with classical PBS ?

Meaning if the parameter set is `WopbsOnly` do we expect the params to work for both the classical pbs key and wopbs key ? 

Also If I remember correctly, the way ou wopbs key is, is that it contains a keyswitch key to go from PBS parameters to Wopbs Parameters, which means that is `WopbsOnly(WopbsParameters)` is meant to be compatible with classical pbs then we'd be doing unecessary keyswitch when calling wopbs

Reply:
I think this is due to the fact we have a "new_wopbs_key_only_for_wopbs" which requires to be able to create a CKS for a WopbsOnly, I'm having a hard time figuring how to organize that differently at the moment as a WopbsOnly key would still need to have a CKS to encrypt stuff even if it won't perform any PBS (with proper dispatch it would just panic on any classic PBS call)

Reply:
wopbs integration, true nightmare stuff

Reply:
should improve once there is only a ServerKey dispatching stuff underneath, as long as you call functions that are legal for a the parameter set used to generate it